internal/datastore/crdb/pool/pool.go

```diff
@@ -309,7 +309,7 @@ func (p *RetryPool) withRetries(ctx context.Context, fn func(conn *pgxpool.Conn)
 		log.Ctx(ctx).Warn().Err(err).Uint8("retries", retries).Msg("error is not resettable or retryable")
 		return err
 	}
-	return &MaxRetryError{MaxRetries: p.maxRetries, LastErr: err}
+	return &MaxRetryError{MaxRetries: maxRetries, LastErr: err}
```

Comment:
so why this change?

Reply:
https://github.com/authzed/spicedb/pull/1537/files#diff-1b011b67a73f910b9841c71da8f83c3c99da6101f476e4d02bfb3d3335eda548R261

this will set the correct maxretries if it's been set to something other than the default (0 in the case of disabled)